flaml/integrations/oai/completion.py

```diff
@@ -650,35 +631,196 @@ def create(cls, context, use_cache=True, **config):
+                },
             ]
             params.pop("prompt", None)
         else:
             params["prompt"] = (
-                prompt.format(**context) if isinstance(prompt, str) else prompt(context)
+                prompt.format(**data_instance)
+                if isinstance(prompt, str)
+                else prompt(data_instance)
             )
-        if use_cache:
-            with diskcache.Cache(cls.cache_path) as cls._cache:
-                return cls._get_response(params)
-        return cls.openai_completion_class.create(**params)
+        return params
+
+    @classmethod
+    def test(
+        cls,
+        data,
+        config,
+        eval_func=None,
+        use_cache=True,
+        agg_method="avg",
+        return_responses_and_per_instance_result=False,
+        seed=41,
+        cache_path=".cache",
+    ):
+        """Evaluate the responses created with the config for the OpenAI API call.
+
+        Args:
+            data (list): The list of test data points.
+            config (dict): Hyperparameter setting for the openai api call.
+            eval_func (Callable): The evaluation function for responses per data instance.
+                The function should take a list of responses and a data point as input,
+                and return a dict of metrics. When not provided (None), we will use the
+                one provided via tune function. Defaults to None.
+
+            ```python
+            def eval_func(responses, **data):
+                solution = data["solution"]
+                success_list = []
+                n = len(responses)
+                for i in range(n):
+                    response = responses[i]
+                    succeed = is_equiv_chain_of_thought(response, solution)
+                    success_list.append(succeed)
+                return {
+                    "expected_success": 1 - pow(1 - sum(success_list) / n, n),
+                    "success": any(s for s in success_list),
+                }
+            ```
+            use_cache (bool, Optional): Whether to use cached responses. Defaults to True.
+            agg_method (str, Callable or a dict of Callable): Result aggregation method (across
+                multiple instances) for each of the metrics. Defaults to 'avg'.
+                An example agg_method in str:
+            ```python
+            agg_method = 'median'
+            ```
+                An example agg_method in a Callable:
+            ```python
+            def my_median(results):
+                return np.median(results)
+            agg_method = my_median
+            ```
+                An example agg_method in a dict of Callable:
+            ```python
+            def my_median(results):
+                return np.median(results)
+            def my_average(results):
+                return np.mean(results)
+            agg_method={'median_success': np.median, 'avg_success': np.mean}
+            ```
```

Comment:
Have you tried to compile the doc? I guess you need to shift the indentation of the code blocks left to make it look well.